includes/class-payment-information.php

```diff
@@ -32,31 +34,31 @@ class Payment_Information {
 	/**
 	 * Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
 	 *
-	 * @var bool
+	 * @var Payment_Initiated_By
 	 */
-	private $off_session;
+	private $payment_initiated_by;
 
 	/**
 	 * Payment information constructor.
 	 *
-	 * @param string            $payment_method The ID of the payment method used for this payment.
-	 * @param \WC_Payment_Token $token The payment token used for this payment.
-	 * @param bool              $off_session Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
+	 * @param string               $payment_method The ID of the payment method used for this payment.
+	 * @param \WC_Payment_Token    $token The payment token used for this payment.
+	 * @param Payment_Initiated_By $payment_initiated_by Indicates whether the payment is merchant-initiated or customer-initiated.
 	 *
 	 * @throws \Exception - If no payment method is found in the provided request.
 	 */
 	public function __construct(
 		string $payment_method,
 		\WC_Payment_Token $token = null,
-		bool $off_session = false
+		Payment_Initiated_By $payment_initiated_by = null
 	) {
 		if ( empty( $payment_method ) && empty( $token ) ) {
 			throw new \Exception( __( 'Invalid payment method. Please input a new card number.', 'woocommerce-payments' ) );
 		}
 
-		$this->payment_method = $payment_method;
-		$this->token          = $token;
-		$this->off_session    = $off_session;
+		$this->payment_method       = $payment_method;
+		$this->token                = $token;
+		$this->payment_initiated_by = $payment_initiated_by ?? Payment_Initiated_By::CUSTOMER();
```

Comment:
Just for added context here: php-enum creates magic, static functions in the enum that return an instantiated class with the right value. That's what the `Payment_Initiated_By::CUSTOMER()` function is about. See the [php-enum readme](https://github.com/myclabs/php-enum#static-methods) for more details.